authorizations/authorization-ranger/src/main/java/org/apache/gravitino/authorization/ranger/RangerAuthorizationPlugin.java

```diff
@@ -0,0 +1,583 @@
+        // idempotent operation.
+        return true;
+      }
+    } else {
+      policy = new RangerPolicy();
+      policy.setService(rangerServiceName);
+      policy.setName(change.getSecurableObject().fullName());
+      policy.setPolicyLabels(Lists.newArrayList(RangerHelper.MANAGED_BY_GRAVITINO));
+
+      List<String> nsMetadataObject =
+          Lists.newArrayList(
+              SecurableObjects.DOT_SPLITTER.splitToList(change.getSecurableObject().fullName()));
+      if (nsMetadataObject.size() > 4) {
+        // The max level of the securable object is `catalog.db.table.column`
+        throw new RuntimeException("The securable object than 4");
+      }
+      nsMetadataObject.remove(0); // remove `catalog`
+
+      List<String> rangerDefinesList =
+          Lists.newArrayList(
+              RangerDefines.RESOURCE_DATABASE,
+              RangerDefines.RESOURCE_TABLE,
+              RangerDefines.RESOURCE_COLUMN);
+      for (int i = 0; i < nsMetadataObject.size(); i++) {
+        RangerPolicy.RangerPolicyResource policyResource =
+            new RangerPolicy.RangerPolicyResource(nsMetadataObject.get(i));
+        policy.getResources().put(rangerDefinesList.get(i), policyResource);
+      }
+    }
+
+    rangerHelper.addPolicyItem(policy, change.getRoleName(), change.getSecurableObject());
+    try {
+      if (policy.getId() == null) {
+        rangerClient.createPolicy(policy);
+      } else {
+        rangerClient.updatePolicy(policy.getId(), policy);
+      }
+    } catch (RangerServiceException e) {
+      throw new RuntimeException(e);
+    }
+
+    return true;
+  }
+
+  /**
+   * Remove the securable object's privilege from the policy. <br>
+   * 1. Find the policy base the metadata object. <br>
+   * 2. If the policy exists and has the same privilege, then remove role name in the policy items.
+   * <br>
+   * 3. If policy does not contain any policy item, then delete this policy. <br>
+   */
+  private boolean doRemoveSecurableObject(RoleChange.RemoveSecurableObject change) {
+    RangerPolicy policy = rangerHelper.findManagedPolicy(change.getSecurableObject());
+    if (policy == null) {
+      LOG.warn(
+          "Cannot find the Ranger policy({}) for the Gravitino securable object({})!",
+          change.getRoleName(),
+          change.getSecurableObject().fullName());
+      // Don't throw exception or return false, because need support immutable operation.
+      return true;
+    }
+
+    policy
+        .getPolicyItems()
+        .forEach(
+            policyItem -> {
+              boolean match =
+                  policyItem.getAccesses().stream()
+                      .allMatch(
+                          // Find the policy item that matches access and role
+                          access -> {
+                            // Use Gravitino privilege to search the Ranger policy item's access
+                            boolean matchPrivilege =
```

Comment:
Could we reuse method `mappingPrivilege` here?

Reply:
They have different logic